src/main/java/no/nav/pto/veilarbportefolje/arbeidsliste/v2/ArbeidsListeV2Controller.java

```diff
@@ -0,0 +1,173 @@
+                .orElse(false);
+
+        Arbeidsliste arbeidsliste = aktoerId
+                .flatMap(arbeidslisteService::getArbeidsliste)
+                .toJavaOptional()
+                .orElse(emptyArbeidsliste())
+                .setIsOppfolgendeVeileder(aktoerId.map(id ->
+                        arbeidslisteService.erVeilederForBruker(fnr, VeilederId.of(innloggetVeileder))).get())
+                .setHarVeilederTilgang(harVeilederTilgang);
+
+        return harVeilederTilgang ? arbeidsliste : emptyArbeidsliste().setHarVeilederTilgang(false);
+    }
+
+    @PostMapping("/arbeidsliste")
+    public Arbeidsliste opprettArbeidsListe(@RequestBody ArbeidslisteV2Request body) {
+        validerOppfolgingOgBruker(body.fnr().get());
+        validerErVeilederForBruker(body.fnr().get());
+        Fnr gyldigFnr = Fnr.ofValidFnr(body.fnr().get());
+        sjekkTilgangTilEnhet(gyldigFnr);
+
+        arbeidslisteService.createArbeidsliste(data(body, gyldigFnr))
+                .onFailure(e -> secureLog.warn("Kunne ikke opprette arbeidsliste: {}", e.getMessage()))
+                .getOrElseThrow((Function<Throwable, RuntimeException>) RuntimeException::new);
+
+        return arbeidslisteService.getArbeidsliste(gyldigFnr).get()
+                .setHarVeilederTilgang(true)
+                .setIsOppfolgendeVeileder(true);
+    }
+
+    @PutMapping("/arbeidsliste")
+    public Arbeidsliste oppdaterArbeidsListe(@RequestBody ArbeidslisteV2Request body) {
+        validerOppfolgingOgBruker(body.fnr().get());
+        Fnr fnr = Fnr.ofValidFnr(body.fnr().get());
+        sjekkTilgangTilEnhet(fnr);
+        validerArbeidslisteV2(body, true);
+
+        arbeidslisteService
+                .updateArbeidsliste(data(body, fnr))
+                .onFailure(e -> secureLog.warn("Kunne ikke oppdatere arbeidsliste: {}", e.getMessage()))
+                .getOrElseThrow((Function<Throwable, RuntimeException>) RuntimeException::new);
+
+        if (arbeidslisteService.getArbeidsliste(fnr).get() == null){
+            VeilederId veilederId = AuthUtils.getInnloggetVeilederIdent();
+            NavKontor enhet = brukerService.hentNavKontor(fnr).orElse(null);
+            secureLog.warn("Arbeidsliste kunne ikke oppdateres, var null, fnr: {}, veileder: {}, på enhet: {}", body.fnr().get(), veilederId, enhet);
+            throw new IllegalStateException("Kunne ikke oppdatere. Fant ikke arbeidsliste for bruker");
+        }
+
+        return arbeidslisteService.getArbeidsliste(fnr).get()
+                .setHarVeilederTilgang(true)
+                .setIsOppfolgendeVeileder(arbeidslisteService.erVeilederForBruker(
+                        fnr,
+                        AuthUtils.getInnloggetVeilederIdent()));
+    }
+
+    @DeleteMapping("/arbeidsliste")
+    public Arbeidsliste deleteArbeidsliste(@RequestBody ArbeidslisteForBrukerRequest arbeidslisteForBrukerRequest) {
```

Comment:
I am not sure that we can send body in delete request...but lets see..

Reply:
Hehe i guess we will find out fast :P